app/assets/javascripts/app/views/feedback_view.js

```diff
@@ -40,7 +43,40 @@ app.views.Feedback = app.views.Base.extend({
     if(evt) { evt.preventDefault(); }
     if(!window.confirm(Diaspora.I18n.t("reshares.post", {name: this.model.reshareAuthor().name}))) { return }
     this.model.interactions.reshare();
-  }
+  },
+
+  blockUser: function(evt) {
+    if(evt) { evt.preventDefault(); }
+    if(!confirm(Diaspora.I18n.t('ignore_user'))) { return; }
+
+    this.model.blockAuthor()
+      .done(function() {
+        // return to stream
+        document.location.href = "/stream";
+      })
+      .fail(function() {
+        Diaspora.page.flashMessages.render({
+          success: false,
+          notice: Diaspora.I18n.t('ignore_failed')
+        });
+      });
+  },
+
+  hidePost : function(evt) {
+    if(evt) { evt.preventDefault(); }
+    if(!confirm(Diaspora.I18n.t('hide_post'))) { return; }
+
+    $.ajax({
+      url : "/share_visibilities/42",
+      type : "PUT",
+      data : {
+        post_id : this.model.id
+      }
+    });
+
+    // return to stream
+    document.location.href = "/stream";
```

Comment:
I implemented this analog to the hidePost in `stream_post_view.js` - maybe it would be good to change that function, too. Remove / Redirect on success, show an error toast if something went wrong?